src/api/category/repository/category.repository.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Category } from '../model/category.model';
+import { Model } from 'mongoose';
+import { CategoryDto } from '../_dto/category.dto';
+
+@Injectable()
+export class CategoryRepository {
+  constructor(
+    @InjectModel(Category.name)
+    private readonly categoryModel: Model<Category>,
+  ) {}
+
+  async findCategory(query: unknown) {
```

Comment:
We are inside Category repo.. So we no need to mention createCategory... just create, findById, findByName.... we can use names like this